python/flatbuffers/builder.py

```diff
@@ -204,32 +206,34 @@ def WriteVtable(self):
 
             vtKey.append(elem)
 
-        vtKey = tuple(vtKey)
-        vt2Offset = self.vtables.get(vtKey)
-        if vt2Offset is None:
+        # Search through existing vtables to see if this layout has already
+        # been written to the buffer. We manually use Python's built-in hash
+        # function in order to avoid storing the actual key, at the cost of
+        # having to subsequently test for equality.
+
+        vtKeyLength = len(vtKey)
```

Comment:
Please choose a more descriptive name than `vtKeyLength`. The reason is that `length` has multiple interpretations in this context: 1) the number of elements in `vtKey`, or 2) the number of bytes in the serialized vtable representation used for comparison.

Reply:
It's literally the length of `vtKey`.

The number of bytes is defined further defined in the same function as `vBytes` (although incidentally this should perhaps be `vtBytes`).

There is no other mention of `length` in this function.

Reply:
@malthe The vtable deduplicator is the most complicated code in the `Builder`, so I'm discussing this within the bigger picture of that. In particular, I'm thinking of making it easy to understand for future contributors.

You'll notice that this line you wrote:

```python
vBytes = vtKeyLength + VtableMetadataFields
```

Interprets `vtKeyLength` as the number of bytes used for vtable. Yet, you also wrote this:

```python
vtKeyLength = len(vtKey)
```

We know that the `current_vtable` is a list of `i32` values, so which is it? Here is how `current_vtable` is initialized elsewhere in the `Builder`, outside of your changes:

```python
        # use 32-bit offsets so that arithmetic doesn't overflow.
        self.current_vtable = [0 for _ in range_func(numfields)]
```

This is the reasoning that leads me to ask you for new variable names :-)

Reply:
That's because of `vBytes *= N.VOffsetTFlags.bytewidth`. This is where it changes from a length to a byte size.

Reply:
@malthe Ah, I see. Please move that multiplication into the definition of vtBytes.

Reply:
Let me clue you in before you get too busy requesting changes. There's a history button that's very useful to see how a codebase changes over time and you'll find a lot of interesting artifacts there including the original definition of `vBytes` which is entirely unchanged here. You're totally out of line here. Feel free to close or do whatever you wish with this PR.

Reply:
@malthe this is entirely normal level of code review. It's not that weird to ask for things that will improve the readability of the code, it benefits everyone. Not sure why this makes you so upset. You'll find the same in most OSS projects you contribute to, in fact, most will be more strict.

Reply:
@rw has probably spent more time reviewing your code than the time it would take him to do it himself. He does that because it makes for a good community, and people can learn from it.

Reply:
This is not about how much time someone has spent or not spent reviewing code, it's about striving to limit the scope of a patch request to the code being changed and contributed. There's always code surrounding the "patch area" – code such as the assignment to `vBytes`. I didn't write this code. It was there when I came to the codebase. Treat OSS contributors with respect and they'll be happy to contribute. Most of them including myself have contributed a great many patches before and are fully aware of the requirements to actually further a project.

Reply:
Again, perfectly normal for it to make sense for surrounding code to be changed too, I certainly don't ever see anyone getting angry over it. I don't see any comment from @rw that I could possibly interpret as disrespectful, which is not something I can say for you.

Reply:
No worries. We can't all agree on everything.